standards/request-response.md

```diff
@@ -444,8 +444,32 @@ Access-Control-Allow-Methods: *
 
 <hr />
 
-```note
-**KEEP AN EYE ON IT**: The [Idempotency-Key Request header](https://tools.ietf.org/id/draft-idempotency-header-01.html) is in an experimental state but getting lots of attention as a pattern of making fault-tolerant resilient requests for traditionally non-idempotent methods like `POST`.
+#### [Idempotency-Key Request header](https://datatracker.ietf.org/doc/html/draft-idempotency-header-01)
+**Type**: Request
+
+**Support**: OPTIONAL
+
+**Description**:
+The HTTP Idempotency-Key request header field can be used to carry an "idempotency key" to make non-idempotent HTTP methods such as POST or PATCH fault-tolerant.
+This is an experimental header, as there is common usage of it, but no accepted spec for it.
+`Idempotency-Key` should typically be a V4 UUID as a string, or another random string with enough entropy to avoid collisions, and 
+should be no longer than 255 characters long.
+An API must in some way persist the Idempotency-Key in a stateful way to manage the idempotency.
+It is common to include the key as part of the primary key or as a secondary index for the resource.
+
+- The header value **MUST** follow this regex format: `[a-z0-9-]{20,255}`
+- An API **MUST** return a `400` when the header value is an invalid format.
+- On conflict, an API **MUST** return a `2xx` if the endpoint is idempotent, or a `409` if the endpoint is not idempotent.
```

Comment:
There's valid reasons that some cases are not idempotent.
The key indicates that you are trying to do some unique "action" with an ID.  There's a lot of cases where an API could assess if the action a client is trying to do is invalid.
In payload-storage-service's case, when a client POSTs a new payload with an Idempotency-Key, this is idempotent, initially.  The client can retry the POST with the same request multiple times, and will always get a 201 back without ever creating duplicate payloads.
But a client uses the same key, but then tries to change the payload content, this is an illegal action.  It's only idempotent as long as the client is uploading the same payload content. A payload content is immutable, so if a naughty API accidentally changed out the payload content from the request every time it retried in a a retry loop, the API needs to indicate that it is doing something that won't work, so it will reject it with a 409.